<cuda-runtime.h> not found while building netlrts-linux-x86_64-cuda builds

Autobuild log: http://charm.cs.illinois.edu/autobuild/old.2019_07_25__01_04/netlrts-linux-x86_64-cuda.txt
```
In file included from ampi.h:11:0,
                 from compat_ampius.C:2:
/scratch/autobuild/netlrts-linux-x86_64-cuda/charm/netlrts-linux-x86_64-cuda/bin/../include/hapi.h:3:26: fatal error: cuda_runtime.h: No such file or directory
 #include <cuda_runtime.h>
                          ^
compilation terminated.
/bin/cp -f sim.h ../../../../include/pose/sim.h
```

--- a/src/arch/common/special.sh
+++ b/src/arch/common/special.sh
@@ -6,7 +6,7 @@ export CHARMINC=../include
 if test -n "$BUILD_CUDA"
 then
   echo "---------------------- special.sh for cuda executing ----------------"
-  if [ "$CMK_GDIR" == "gni" ] ; then
+  if [ "$CMK_GDIR" = "gni" ] ; then
     export CRAY_CUDA_PROXY=1
   fi
 

--- a/src/libs/ck-libs/ampi/Makefile
+++ b/src/libs/ck-libs/ampi/Makefile
@@ -1,6 +1,11 @@
 CDIR := $(shell cd ../../../.. && pwd)
 -include $(CDIR)/include/conv-mach-opt.mak
-CHARMC=$(CDIR)/bin/charmc $(OPTS)
+
+ifeq ($(BUILD_CUDA),1)
+CUDA_INCLUDE=-I$(CUDA_DIR)/include
+endif
+
+CHARMC=$(CDIR)/bin/charmc $(OPTS) $(CUDA_INCLUDE)
 
 HEADERS=ampi.h ampif.h mpio.h mpiof.h ampiimpl.h \
 		ampi-interoperate.h ampi.decl.h \

--- a/src/libs/ck-libs/ampi/ampiCC
+++ b/src/libs/ck-libs/ampi/ampiCC
@@ -80,6 +80,10 @@ Do() {
   eval "$@"
 }
 
+if [ $BUILD_CUDA -eq 1 ]; then
+  ARGS="$ARGS -I$CUDA_DIR/include"
+fi
+
 if [ -n "$FSGLOBALS" ]
 then
   if [ "$CMK_SUPPORTS_FSGLOBALS" != '1' ]

--- a/src/scripts/charmc
+++ b/src/scripts/charmc
@@ -1420,6 +1420,11 @@ fi
 if [ -n "$AMPI" ]
 then
   CMK_SYSINC="-I$CHARMINC/ampi $CMK_SYSINC"
+  if [ "$BUILD_CUDA" -eq 1 ]; then
+    # If Charm++ is built with CUDA support, all AMPI applications need to be built with CUDA
+    POST_LIBRARIES="$POST_LIBRARIES -L$CUDA_DIR/lib64 -lcudart" # For normal build
+    USER_POST_LIBRARIES="$USER_POST_LIBRARIES -L$CUDA_DIR/lib64 -lcudart" # For -standalone build
+  fi
 fi
 
 if [ "$TLSGLOBALS" = "1" ]

